# 과제 1

In [ ]:
import pandas as pd 
df_t = pd.read_csv('/content/drive/MyDrive/ADP24/class101/2.데이터_핸들링/data/titanic.csv')

In [ ]:
df_t

survived  pclass     sex   age  ...  embarked   class  adult_male  embark_town
0           0       3    male  22.0  ...         S   Third        True  Southampton
1           1       1  female  38.0  ...         C   First       False    Cherbourg
2           1       3  female  26.0  ...         S   Third       False  Southampton
3           1       1  female  35.0  ...         S   First       False  Southampton
4           0       3    male  35.0  ...         S   Third        True  Southampton
..        ...     ...     ...   ...  ...       ...     ...         ...          ...
886         0       2    male  27.0  ...         S  Second        True  Southampton
887         1       1  female  19.0  ...         S   First       False  Southampton
888         0       3  female   NaN  ...         S   Third       False  Southampton
889         1       1    male  26.0  ...         C   First        True    Cherbourg
890         0       3    male  32.0  ...         Q   Third        True   Queenstown

[891 rows x 11 columns]

In [ ]:
# Q1. pandas를 import하고 data폴더 안에 있는 titanic.csv 파일 불러오세요

In [ ]:
import pandas as pd 
df_t = pd.read_csv('/content/drive/MyDrive/ADP24/class101/2.데이터_핸들링/data/titanic.csv')
df_t

survived  pclass     sex   age  ...  embarked   class  adult_male  embark_town
0           0       3    male  22.0  ...         S   Third        True  Southampton
1           1       1  female  38.0  ...         C   First       False    Cherbourg
2           1       3  female  26.0  ...         S   Third       False  Southampton
3           1       1  female  35.0  ...         S   First       False  Southampton
4           0       3    male  35.0  ...         S   Third        True  Southampton
..        ...     ...     ...   ...  ...       ...     ...         ...          ...
886         0       2    male  27.0  ...         S  Second        True  Southampton
887         1       1  female  19.0  ...         S   First       False  Southampton
888         0       3  female   NaN  ...         S   Third       False  Southampton
889         1       1    male  26.0  ...         C   First        True    Cherbourg
890         0       3    male  32.0  ...         Q   Third        True   Queenstown

[891 rows x 11 columns]

In [ ]:
# Q2. 타이타닉 데이터의 결측치가 있는 변수를 확인하세요 

In [ ]:
df_t.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   survived     891 non-null    int64  
 1   pclass       891 non-null    int64  
 2   sex          891 non-null    object 
 3   age          714 non-null    float64
 4   sibsp        891 non-null    int64  
 5   parch        891 non-null    int64  
 6   fare         891 non-null    float64
 7   embarked     889 non-null    object 
 8   class        891 non-null    object 
 9   adult_male   891 non-null    bool   
 10  embark_town  889 non-null    object 
dtypes: bool(1), float64(2), int64(4), object(4)
memory usage: 70.6+ KB


## 총 891개 인덱스 중에, 아래 3개의 컬럼에서 Null 이 확인됨
- age
- embarked
- embark_town

In [ ]:
# Q3. 타이타닉 데이터에서 survived 변수 (생존 여부) 에서 생존자 수를 구하세요 

In [ ]:
df_t['survived'].value_counts()

0    549
1    342
Name: survived, dtype: int64

## survived 가 1로 기록된 것이 342개, 342명임

In [ ]:
# Q4. 타이타닉 데이터에 class변수의 unique 값을 구하세요

In [ ]:
df_t['class'].unique()

array(['Third', 'First', 'Second'], dtype=object)

## class 컬럼의 고유값은 아래 3개임
- First
- Second
- Third

In [ ]:
# Q5. 타이타닉 데이터에 class변수의 value_counts값을 출력하세요 

In [ ]:
df_t['class'].value_counts()

Third     491
First     216
Second    184
Name: class, dtype: int64

In [ ]:
# Q6. 타이타닉 데이터에서 class변수명을 CLASS로 변경하세요 

In [ ]:
df_t.rename({"class":"CLASS"},axis=1,inplace=True)
df_t

survived  pclass     sex   age  ...  embarked   CLASS  adult_male  embark_town
0           0       3    male  22.0  ...         S   Third        True  Southampton
1           1       1  female  38.0  ...         C   First       False    Cherbourg
2           1       3  female  26.0  ...         S   Third       False  Southampton
3           1       1  female  35.0  ...         S   First       False  Southampton
4           0       3    male  35.0  ...         S   Third        True  Southampton
..        ...     ...     ...   ...  ...       ...     ...         ...          ...
886         0       2    male  27.0  ...         S  Second        True  Southampton
887         1       1  female  19.0  ...         S   First       False  Southampton
888         0       3  female   NaN  ...         S   Third       False  Southampton
889         1       1    male  26.0  ...         C   First        True    Cherbourg
890         0       3    male  32.0  ...         Q   Third        True   Queenstown

[891 rows x 11 columns]

In [ ]:
# Q7. 타이타닉 데이터에 age 변수를를 int로 변경하세요 

In [ ]:
df_t['age'] = df_t['age'].fillna(0).astype(int)
df_t.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   survived     891 non-null    int64  
 1   pclass       891 non-null    int64  
 2   sex          891 non-null    object 
 3   age          891 non-null    int64  
 4   sibsp        891 non-null    int64  
 5   parch        891 non-null    int64  
 6   fare         891 non-null    float64
 7   embarked     889 non-null    object 
 8   CLASS        891 non-null    object 
 9   adult_male   891 non-null    bool   
 10  embark_town  889 non-null    object 
dtypes: bool(1), float64(1), int64(5), object(4)
memory usage: 70.6+ KB


In [ ]:
# Q8. embarked변수를 index로 변경하세요 

In [ ]:
df_t.set_index('embarked',inplace=True)
df_t

survived  pclass     sex  ...   CLASS  adult_male  embark_town
embarked                            ...                                 
S                0       3    male  ...   Third        True  Southampton
C                1       1  female  ...   First       False    Cherbourg
S                1       3  female  ...   Third       False  Southampton
S                1       1  female  ...   First       False  Southampton
S                0       3    male  ...   Third        True  Southampton
...            ...     ...     ...  ...     ...         ...          ...
S                0       2    male  ...  Second        True  Southampton
S                1       1  female  ...   First       False  Southampton
S                0       3  female  ...   Third       False  Southampton
C                1       1    male  ...   First        True    Cherbourg
Q                0       3    male  ...   Third        True   Queenstown

[891 rows x 10 columns]